doc/source/api/core/typing.rst

```diff
@@ -76,41 +76,34 @@ pyvista.TransformLike
 .. autodata:: TransformLike
 
 
-Concrete Data Object Types
---------------------------
-
-pyvista.ConcreteGridType
-~~~~~~~~~~~~~~~~~~~~~~~~
-
-.. currentmodule:: pyvista
-
-.. autotypevar:: ConcreteGridType
+Data Object Types
```

Comment:
Now that the type definitions are much simpler I'm not so sure if these types should be published anymore. 

The main reason to include types here is to make them auto-linked. But this only works in docstrings, not function signatures (the only place these types are likely to be used)

Plus, the names I think are now too similar to the actual classes, and so having `pyvista.DataSetType` in the namespace alongside `pyvista.DataSet` is not ideal.

Suggest removing all types from the docs and removing from the `pyvista` namespace. Can probably move the types to their respective modules now, e.g. move `DataSetType` to `core.dataset.py`